core/src/main/java/io/questdb/std/Decimal128.java

```diff
@@ -0,0 +1,2174 @@
+        } else {
+            // Complex case: full 128-bit conversion
+            fullToSink(sink);
+        }
+    }
+
+    /**
+     * Returns a string representation of this decimal value.
+     *
+     * @return string representation of this Decimal128
+     */
+    @Override
+    public String toString() {
+        // Use StringSink which is already a CharSink - for compatibility
+        StringSink sink = new StringSink();
+        toSink(sink);
+        return sink.toString();
+    }
+
+    /**
+     * Generic function to make a 128-bit addition.
+     *
+     * @param result Decimal128 that will store the result of the operation
+     * @param aH     High 64-bit part of the first operand.
+     * @param aL     Low 64-bit part of the first operand.
+     * @param aScale Scale of the first operand.
+     * @param bH     High 64-bit part of the second operand.
+     * @param bL     Low 64-bit part of the second operand.
+     * @param bScale Scale of the second operand.
+     */
+    private static void add(Decimal128 result, long aH, long aL, int aScale, long bH, long bL, int bScale) {
+        result.scale = aScale;
+        if (aScale < bScale) {
+            // We need to rescale a to the same scale as b
+            result.high = aH;
+            result.low = aL;
+            result.rescale(bScale);
+            aH = result.high;
+            aL = result.low;
+        } else if (aScale > bScale) {
+            // We need to rescale b to the same scale as a
+            result.high = bH;
+            result.low = bL;
+            result.scale = bScale;
+            result.rescale(aScale);
+            bH = result.high;
+            bL = result.low;
+        }
+
+        // Perform 128-bit addition
+        long sumLow = aL + bL;
+
+        // Check for carry
+        long carry = hasCarry(aL, bL, sumLow) ? 1 : 0;
+
+        result.low = sumLow;
+        result.high = Math.addExact(aH, Math.addExact(bH, carry));
+        result.updateCompact();
+    }
+
+    /**
+     * Append a long value to sink without allocation
+     */
+    private static void appendLongToSink(long value, CharSink<?> sink) {
+        if (value == 0) {
+            sink.putAscii('0');
+            return;
+        }
+
+        // Find the highest power of 10 that fits in the value
+        long divisor = 1;
+        long temp = value;
+        while (temp >= 10) {
+            divisor *= 10;
+            temp /= 10;
+        }
+
+        // Output digits from most significant to least significant
+        while (divisor > 0) {
+            int digit = (int) (value / divisor);
+            sink.putAscii((char) ('0' + digit));
+            value %= divisor;
+            divisor /= 10;
+        }
+    }
+
+    private static long calculatePowerOf10(int n) {
+        if (n <= 18) {
+            return LONG_TEN_POWERS_TABLE[n];
+        }
+        long result = 1;
+        while (n >= 18) {
+            result *= LONG_TEN_POWERS_TABLE[18];
```

Comment:
if we could not store powers of ten in the `LONG_TEN_POWERS_TABLE` because it would overflow, how would `result` not overflow ?